Fix build errors related to wallet service

```
> strukly-backend@1.0.0 build
> tsc

src/infrastructure/controllers/wallet_controllers.ts:9:46 - error TS2345: Argument of type 'WalletRepositoryPrisma' is not assignable to parameter of type 'IWalletRepository'.
  Types of property 'findByUser' are incompatible.
    Type '(userId: string) => Promise<Wallet[]>' is not assignable to type '(userId: UserID) => Promise<Wallet[]>'.
      Types of parameters 'userId' and 'userId' are incompatible.
        Type 'UserID' is not assignable to type 'string'.

9 const createWalletUseCase = new CreateWallet(walletRepository);
                                               ~~~~~~~~~~~~~~~~

src/infrastructure/repositories/prisma_wallet_repository.ts:23:33 - error TS2339: Property 'amount' does not exist on type 'Money'.

23         balance: wallet.balance.amount,
                                   ~~~~~~

src/infrastructure/repositories/prisma_wallet_repository.ts:51:9 - error TS2416: Property 'findByUser' in type 'WalletRepositoryPrisma' is not assignable to the same property in base type 'IWalletRepository'.
  Type '(userId: string) => Promise<Wallet[]>' is not assignable to type '(userId: UserID) => Promise<Wallet[]>'.
    Types of parameters 'userId' and 'userId' are incompatible.
      Type 'UserID' is not assignable to type 'string'.

51   async findByUser(userId: string): Promise<Wallet[]> {
           ~~~~~~~~~~

src/infrastructure/repositories/prisma_wallet_repository.ts:70:33 - error TS2339: Property 'amount' does not exist on type 'Money'.

70         balance: wallet.balance.amount,
                                   ~~~~~~


Found 4 errors in 2 files.

Errors  Files
     1  src/infrastructure/controllers/wallet_controllers.ts:9
     3  src/infrastructure/repositories/prisma_wallet_repository.ts:23
```

--- a/strukly-backend/src/infrastructure/repositories/prisma_wallet_repository.ts
+++ b/strukly-backend/src/infrastructure/repositories/prisma_wallet_repository.ts
@@ -6,6 +6,7 @@ import { IWalletRepository } from "../../domain/repositories/wallet_repository";
 import Money from "../../domain/values/money";
 import TransactionMethod from "../../domain/values/transaction_method";
 import WalletID from "../../domain/values/wallet_id";
+import UserID from "src/domain/values/user_id";
 
 export default class WalletRepositoryPrisma implements IWalletRepository {
   private prisma: PrismaClient;
@@ -20,7 +21,7 @@ export default class WalletRepositoryPrisma implements IWalletRepository {
         id: wallet.id.value,
         userID: wallet.userID,
         name: wallet.walletName,
-        balance: wallet.balance.amount,
+        balance: wallet.balance.value,
         transaction_method: wallet.transactionMethod.value,
       },
     });
@@ -48,9 +49,9 @@ export default class WalletRepositoryPrisma implements IWalletRepository {
     });
   }
 
-  async findByUser(userId: string): Promise<Wallet[]> {
+  async findByUser(userId: UserID): Promise<Wallet[]> {
     const wallets = await this.prisma.wallet.findMany({
-      where: { userID: userId },
+      where: { userID: userId.value },
     });
     
     const result = wallets.map((w) => new Wallet({
@@ -67,7 +68,7 @@ export default class WalletRepositoryPrisma implements IWalletRepository {
     const updated = await this.prisma.wallet.update({
       where: { id: wallet.id.value },
       data: {
-        balance: wallet.balance.amount,
+        balance: wallet.balance.value,
       },
     });
     return new Wallet({

